src/VisualStudio/Core/Def/Storage/FileDownloader.cs

```diff
@@ -30,9 +30,36 @@ public IFileDownloader CreateClient(string hostId, string serverPath, int pollin
     private FileDownloader(RemoteControlClient client)
         => _client = client;
 
+#if NET
+
+    /// <summary>
+    /// The netcore version of <see cref="RemoteControlClient"/> doesn't support ReturnStale.  It will download the file
+    /// (on a separate thread), but then not cache it because it doesn't have access to the normal IE component that
+    /// does proper header reading/caching.  Then, when we call in to read the file, we get nothing back, since nothing
+    /// was cached.
+    /// <para/> The temporary solution to this is to force the download to happen.  This is not ideal as we will no
+    /// longer be respecting the server "Cache-Control:Max-Age" header.  Which means we'll continually download the
+    /// files, even if not needed (since the server says to use the local value).  This is not great, but is not
+    /// terrible either.  First, we will only download the full DB file <em>once</em>, when it is actually missing on
+    /// the user's machine.  From that point on, we'll only be querying the server for the delta-patch file for the DB
+    /// version we have locally.  The vast majority of the time that is a tiny document of the form <c><![CDATA[<Patch
+    /// upToDate="true" FileVersion="105" ChangesetId="1CBE1453" />]]></c> (around 70 bytes) which simply tells the user
+    /// they are up to date.  Only about once every three months will they actually download a large patch file.  Also,
+    /// this patch download will only happen once a day tops (as that is our cadence for checking if there are new index
+    /// versions out).
+    /// <para/> https://github.com/dotnet/roslyn/issues/71014 tracks this issue.  Once RemoteControlClient is updated to
+    /// support this again, we can remove this specialized code for netcore.
+    /// </summary>
+    public Task<Stream> ReadFileAsync()
+        => _client.ReadFileAsync(BehaviorOnStale.ForceDownload);
```

Comment:
What happens right now when we use ReturnStale? Does the library immediately reject it (since it doesn't support it?) or does it do something bad instead? If it's immediately rejected can we instead have this fall back so that way when the support is added it lights up rather than having to remember to delete this?

Reply:
it literally always retursn null because it never has anything in the cache.  so we're 100% broken.  

i don't mind the fallback appraoch.  i can switch to that.